src/sentry/ownership/grammar.py

```diff
@@ -158,7 +147,12 @@ def test_frames(
             glob_match(val, pattern, ignorecase=True, path_normalize=True)
         ),
     ) -> bool:
-        for frame in (f for f in _iter_frames(data) if isinstance(f, Mapping)):
+        platform = data.get("platform")
+        frames = find_stack_frames(data)
+        if platform:
+            supplement_filename(platform, frames)
```

Comment:
This is pretty much the most interesting part. Not a whole lot to look at for this PR. 

Added tests to cover scenarios that this change introduces. Swapping out `_iter_frames` for `find_stack_frames` did not cause any previous tests.

Reply:
I'm actually wondering here if we should match both the supplemented filename and the original one (assuming they're different).

Mostly concerned that users might already have rules in place that expect the absolute filenames and might start failing with the supplemented ones.

We might also not want to do this at this level - Otherwise we will repeat this logic for every rule in an ownership file: https://github.com/getsentry/sentry/blob/c102de5439edc10a73a84b6641d6410e8e7d3e8b/src/sentry/models/projectownership.py#L195-L202

I also realised that if we modify the event data in place we're going to potentially affect other places that use this data. I'm wondering if it makes sense to:
 - `deepcopy` the data before we pass it into the rule matchers
 - When supplementing the filename, give it a separate name so that we can match against both fields (if different)

Reply:
Ahh, you're right - forgot about your note about using a different name for the munged path. Hmm ...